ui/src/app/views/homepage/component.ts

```diff
@@ -127,13 +134,27 @@ export class HomepageComponent implements OnInit, OnDestroy {
   addWorkspace(): void {
     this.router.navigate(['workspace/build'], {relativeTo : this.route});
   }
-   navigateToProfile(): void {
-    this.router.navigate(['profile']);
-   }
 
-   listWorkspaces(): void {
-    this.router.navigate(['workspaces']);
-   }
+  navigateToProfile(): void {
+   this.router.navigate(['profile']);
+  }
+
+  listWorkspaces(): void {
+   this.router.navigate(['workspaces']);
+  }
+
+  hasCdrAccess(): boolean {
+    if (!this.enforceRegistered) {
+      return true;
+    }
+    if (!this.profile) {
```

Comment:
I think we need to move this above `!this.enforceRegistered` or currently it will succeed if you don't have a profile yet, if enforce registered is disabled.

Reply:
I was treating "hasCdrAccess" as orthogonal to the account initialization state so this was returning the right thing - but changed per above comment to remove FCAccountInitialized.

For reference, here are the effective predicates for the two buttons on screen:
- List workspaces: `FCAccountInitialized && (!enforceRegistered || dataAccess >= registered)`
- Add workspace: `BillingProjectReady && (!enforceRegistered || dataAccess >= registered)`

Reply:
Looks great! Yeah, that looks good on the predicates.